### PR TITLE
Fixing APPCLOUD-298 Changing the version format

### PR DESCRIPTION
**Pull request by kasundsilva**

### CODE REVIEW

File: modules/jaggeryapps/appmgt/src/site/blocks/application/application.jag [hunk 1]
```diff
@@ -628,16 +628,10 @@ function validationApplicationRevision(field, value) {
...
-        return true;
-    }
-
-    var patternNonAlphaNum = /^(?:(\d{1,2})\.)?(?:(\d{1,2})\.)?(\*|\d{1,2})$/g;
-    if (!patternNonAlphaNum.test(value)) {
+    var patternNonAlphaNum = /[^a-zA-Z\d\-\_.]/g;
+    if (patternNonAlphaNum.test(value)) {
         response.status = 400;
-        response.content = "Bad request : Invalid version number - Provide version number in format major.minor.patch";
+        response.content = "Bad request : Application Name has invalid characters. Valid characters are [a-z, A-Z, 0-9, _, ., -]";
```
Comment (SabraO) [inline modules/jaggeryapps/appmgt/src/site/blocks/application/application.jag]: Review text

File: modules/jaggeryapps/appmgt/src/site/themes/default/templates/application/template.jag [hunk 2]
```diff
@@ -624,11 +625,10 @@ jagg.template("application", function (inputs, outputs, jagg) {
         }
 
         function validationApplicationRevisionChar(value) {
-            var patternNonAlphaNum = /^(?:(\d{1,2})\.)?(?:(\d{1,2})\.)?(\*|\d{1,2})$/g;
+            var patternNonAlphaNum = /[^a-zA-Z\d\-\_.]/g;
```
Comment (SabraO) [inline modules/jaggeryapps/appmgt/src/site/themes/default/templates/application/template.jag]: add a proper variable name

File: modules/jaggeryapps/appmgt/src/site/themes/default/templates/application/template.jag [hunk 3]
```diff
@@ -696,6 +696,11 @@ jagg.template("application", function (inputs, outputs, jagg) {
         }
         var appNameField = $("#applicationName");
         var createNewAppForm = $("#application-details");
+   /*     var appVerionField = $("#applicationRevision");
```
Comment (SabraO) [inline modules/jaggeryapps/appmgt/src/site/themes/default/templates/application/template.jag]: Remove unused function

File: modules/jaggeryapps/appmgt/src/site/themes/default/templates/application/template.jag [hunk 4]
```diff
@@ -778,6 +783,20 @@ jagg.template("application", function (inputs, outputs, jagg) {
         }
     }
 
+    function checkApplicationVersionExistence(version){
+        var existingVersions = versionList.split(",");
+        var version = version.toLowerCase() ;
+        var verionMatch;
+        for (var i = 0; i < existingVersions.length; i++){
+            var existingVerion = existingVersions[i].toLowerCase();
+            verionMatch = existingVerion.localeCompare(version.toLowerCase());
```
Comment (SabraO) [inline modules/jaggeryapps/appmgt/src/site/themes/default/templates/application/template.jag]: use indexof here.

File: modules/jaggeryapps/appmgt/src/site/themes/default/templates/application/template.jag (hunk 2 above)
Comment (kasundsilva) [inline modules/jaggeryapps/appmgt/src/site/themes/default/templates/application/template.jag]: Fixed https://github.com/wso2/app-cloud/pull/659

File: modules/jaggeryapps/appmgt/src/site/themes/default/templates/application/template.jag (hunk 3 above)
Comment (kasundsilva) [inline modules/jaggeryapps/appmgt/src/site/themes/default/templates/application/template.jag]: Fixed

File: modules/jaggeryapps/appmgt/src/site/themes/default/templates/application/template.jag (hunk 4 above)
Comment (kasundsilva) [inline modules/jaggeryapps/appmgt/src/site/themes/default/templates/application/template.jag]: Done https://github.com/wso2/app-cloud/pull/699

**Pull request merged**